### ISSUE TITLE
Automatically fix binary directory

### ISSUE THREAD

**Issue by jakob** [PostgresApp/PostgresApp#708]
Since making Postgres.app relocatable, we've received some reports about "PostgreSQL Version not installed".

This is something that would happen when people eg. start Postgres.app from the disk image, and ignore the warning. Postgres.app would add a new server, which would then end up with a broken binary directory (the binary directory would have some weird path due to app translocation).

The error message is somewhat confusing, since the problem isn't the missing PostgreSQL version, but rather the incorrect binary path.

So my suggestion would be for Postgres.app to automatically update the binary path if it detects that it points to a non-existing directory.

I think updating the path should be safe given the following conditions:

1. The binary directory points to a non-existing path
2. The current bundle contains the PostgreSQL binaries for the same major version
3. The data directory was never started with a newer minor version

If we do that, then Postgres.app will "just work" in most situations. I can't think of a situation where this would break a users working setup since it would only fix broken configurations.

It would even allow users to run Postgres.app directly from the disk image, which would be a nice-to-have feature.

**Comment by jakob** [comment]
I've made a first attempt to implement this in ca016624f8b6df47c2c92a1cb59a9d189bfc7521.

Here's how I implemented it:

When the app checks for the binaries directory, and the directory does not exist, it does the following:
   1. Remove everything up to ".app/" in the binary path
   2. Check if the currently running app contains binaries with the same binary path
   3. When starting the server, it automatically updates the binary server if necessary

This means that the binary path is updated automatically when the user moves or renames the app. It should now even work for launching the app from the disk image (the effect of App Translocation is similar to moving the app).

The login helper / menu helpers currently do not automatically update the binary location.
